plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js

```diff
@@ -23,6 +25,28 @@ export default {
 
     this.siteSettings = container.lookup("service:site-settings");
 
+    withPluginApi("1.8.0", (api) => {
+      api.addSidebarPanel(
+        (BaseCustomSidebarPanel) =>
+          class ChatSidebarPanel extends BaseCustomSidebarPanel {
+            key = "chat";
+            switchButtonLabel = I18n.t("sidebar.panels.chat.label");
+            switchButtonIcon = "d-chat";
+            switchButtonDefaultUrl = getURL("/chat");
+          }
+      );
+
+      const currentUser = api.getCurrentUser();
+      if (getUserChatSeparateSidebarMode(currentUser).fullscreen) {
+        api.setCombinedSidebarMode();
+        api.showSidebarSwitchPanelButtons();
+      } else if (getUserChatSeparateSidebarMode(currentUser).always) {
+        api.setSeparatedSidebarMode();
+      } else {
+        api.hideSidebarSwitchPanelButtons();
+      }
+    });
```

Comment:
This will be a bit easier to read if we introduce a variable:

```suggestion
      const currentUser = api.getCurrentUser();
      const chatSeparateSidebarMode = getUserChatSeparateSidebarMode(currentUser);
      
      if (chatSeparateSidebarMode.fullscreen) {
        api.setCombinedSidebarMode();
        api.showSidebarSwitchPanelButtons();
      } else if (chatSeparateSidebarMode.always) {
        api.setSeparatedSidebarMode();
      } else {
        api.hideSidebarSwitchPanelButtons();
      }
    });
```